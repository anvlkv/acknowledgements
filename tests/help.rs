use std::path::PathBuf;
use std::process::Command;

#[test]
fn print_help() {
    let dot = std::env::var("CARGO_MANIFEST_DIR").unwrap();
    let mut path = PathBuf::from(&dot);
    path.push("target/debug/acknowledge");
    let output = Command::new(path)
        .arg(format!("--help"))
        .output()
        .expect("Failed to run");
    let printed = String::from_utf8(output.stdout).expect("Failed to parse");

    insta::assert_snapshot!(printed);
}
