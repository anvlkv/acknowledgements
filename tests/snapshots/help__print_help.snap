---
source: tests/help.rs
expression: printed
---
acknowledge is a simple CLI tool to analyze dependencies of a Cargo (rust) project and produce an ACKNOWLEDMENTS.md file listing (major) contributors of your dependencies

Usage: acknowledge [OPTIONS] --path <PATH> [COMMAND]

Commands:
  clear-cache  Clears cache
  help         Print this message or the help of the given subcommand(s)

Options:
  -p, --path <PATH>
          Path to Cargo project for analysis
  -g, --gh-token <GH_TOKEN>
          Running Acknowledgements on any project of reasonable size you're likely to face rate limits. Please provide a personal access token
  -o, --output <OUTPUT>
          Output file path, defaults to project path if not provided
  -m, --mention
          Whether to include @ (at) symbol in front of a github user's name
  -f, --format <FORMAT>
          Format of the output file [default: NameAndCount]
  -b, --breadth <BREADTH>
          Breadth of scan, whether to include optional, build and dev deps contributors [default: NonOpt]
  -c, --contributions-threshold <CONTRIBUTIONS_THRESHOLD>
          Min number of contributions to be included in the list, doesn't apply to sole contributors [default: 2]
  -s, --sources <SOURCES>
          List other sources, not specified in Cargo.toml
  -t, --template <TEMPLATE>
          Use your own template. See https://github.com/anvlkv/acknowledgements/blob/main/src/template.md?plain=1 for reference
  -h, --help
          Print help
  -V, --version
          Print version
